packages/wallet/src/test-utils/launchNode.ts

```diff
@@ -0,0 +1,133 @@
+  consensusKey,
+  dbPath,
+  args = defaultFuelCoreArgs,
+  useSystemFuelCore = false,
+}: LaunchNodeOptions): Promise<() => void> =>
+  new Promise((resolve) => {
+    // This string is logged by the client when the node has successfully started. We use it to know when to resolve.
+    const graphQLStartSubstring = 'Binding GraphQL provider to';
+
+    const command = useSystemFuelCore ? 'fuel-core' : './node_modules/.bin/fuels-core';
+
+    const child = spawn(command, [
+      'run',
+      '--db-path',
+      dbPath, // hardcoded
+      '--consensus-key',
+      consensusKey,
+      '--chain',
+      chainConfigPath,
+      ...args,
+    ]);
+
+    // Cleanup function where fuel-core is stopped.
+    const cleanup = () => {
+      kill(Number(child.pid));
+
+      // Remove all the listeners we've added.
+      child.stdout.removeAllListeners();
+      child.stderr.removeAllListeners();
+    };
+
+    child.stderr.setEncoding('utf8');
+
+    // Look for a specific graphql start point in the output.
+    child.stderr.on('data', (chunk: string) => {
+      // Look for the graphql service start.
+      if (chunk.indexOf(graphQLStartSubstring) !== -1) {
+        // Resolve with the cleanup method.
+        resolve(cleanup);
+      }
+    });
+
+    // Process exit.
+    process.on('exit', cleanup);
+
+    // Catches ctrl+c event.
+    process.on('SIGINT', cleanup);
+
+    // Catches "kill pid" (for example: nodemon restart).
+    process.on('SIGUSR1', cleanup);
+    process.on('SIGUSR2', cleanup);
+
+    // Catches uncaught exceptions.
+    process.on('uncaughtException', cleanup);
+  });
+
+const generateWallets = async (count: number, provider: Provider) => {
+  const wallets: WalletUnlocked[] = [];
+  for (let i = 0; i < count; i += 1) {
+    const wallet = await generateTestWallet(provider, [[1_000, BaseAssetId]]);
+    wallets.push(wallet);
+  }
+  return wallets;
+};
+
+/**
+ * Launches a fuel-core node and returns a provider, 10 wallets, and a cleanup function to stop the node.
+ * @param launchNodeOptions - options to launch the fuel-core node with.
+ * @param walletCount - the number of wallets to generate. (optional, defaults to 10)
+ * */
+export const launchNodeAndGetWallets = async ({
+  launchNodeOptions,
+  walletCount = 10,
+}: {
+  launchNodeOptions?: Partial<LaunchNodeOptions>;
+  walletCount?: number;
+} = {}) => {
+  // Write a temporary chain configuration file.
+  await fs.writeFile('.chainConfig.json', JSON.stringify(defaultChainConfig), 'utf8');
+
+  // Create a temp db directory.
+  await fs.mkdir('.fuel-core-db', { recursive: true });
+
+  const defaultNodeOptions: LaunchNodeOptions = {
+    chainConfigPath: '.chainConfig.json',
+    consensusKey: '0xa449b1ffee0e2205fa924c6740cc48b3b473aa28587df6dab12abc245d1f5298',
+    dbPath: '.fuel-core-db',
+  };
+
+  const closeNode = await launchNode({ ...defaultNodeOptions, ...launchNodeOptions });
+
+  const provider = new Provider('http://127.0.0.1:4000/graphql');
```

Comment:
As you launch a node, the return could include the `ip` and `port` instead of having fixed one.

Reply:
And for the default params you can use the `0.0.0.0` as IP and for the port you can get the next available port on from the OS.